numpy/core/src/umath/loops.c.src

```diff
@@ -1619,6 +1619,31 @@ TIMEDELTA_mm_m_remainder(char **args, npy_intp *dimensions, npy_intp *steps, voi
     }
 }
 
+NPY_NO_EXPORT void
+TIMEDELTA_mm_q_floor_divide(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
+{
+    BINARY_LOOP {
+        const npy_timedelta in1 = *(npy_timedelta *)ip1;
+        const npy_timedelta in2 = *(npy_timedelta *)ip2;
+        if (in1 == NPY_DATETIME_NAT ||
+            in2 == NPY_DATETIME_NAT ||
+            in2 == 0) {
+            npy_set_floatstatus_divbyzero();
+            *((npy_longlong *)op1) = 0;
+        }
+        else {
+            const npy_longlong in1 = *(npy_longlong *)ip1;
+            const npy_longlong in2 = *(npy_longlong *)ip2;
+            if (((in1 > 0) != (in2 > 0)) && (in1 % in2 != 0)) {
+                *((npy_longlong *)op1) = in1/in2 - 1;
+            }
+            else {
+                *((npy_longlong *)op1) = in1/in2;
```

Comment:
Better to use `npy_timedelta` for all these casts, like we do in all the other loops

Reply:
Good point, done. That also allows us to remove two spurious casting lines above.

Reply:
This comment was wrong, apologies